src/Sylius/Behat/Context/Ui/Shop/CartContext.php

```diff
@@ -0,0 +1,365 @@
+    }
+
+    /**
+     * @When  I see the summary of my cart
+     */
+    public function iOpenCartSummaryPage()
+    {
+        $this->summaryPage->open();
+    }
+
+    /**
+     * @Then my cart should be empty
+     */
+    public function iShouldBeNotifiedThatMyCartIsEmpty()
+    {
+        Assert::contains(
+             $this->summaryPage->getInformationAboutEmptyCart(),
+            'Your cart is empty',
+            'There should appear information about empty cart, but it does not.'
+        );
+    }
+
+    /**
+     * @Given /^I (?:remove|removed) product "([^"]+)" from the cart$/
+     */
+    public function iRemoveProductFromTheCart($productName)
+    {
+        $this->summaryPage->open();
+        $this->summaryPage->removeProduct($productName);
+    }
+
+    /**
+     * @Given I change :productName quantity to :quantity
+     */
+    public function iChangeQuantityTo($productName, $quantity)
+    {
+        $this->summaryPage->open();
+        $this->summaryPage->changeQuantity($productName, $quantity);
+    }
+
+    /**
+     * @Then grand total value should be :total
+     * @Then my cart total should be :total
+     */
+    public function myCartTotalShouldBe($total)
+    {
+        $this->summaryPage->open();
+        Assert::eq(
+            $this->summaryPage->getGrandTotal(),
+            $total,
+            'Grand total should be %2$s, but it is %s.'
+        );
+    }
+
+    /**
+     * @Then tax total value should be :taxTotal
+     * @Then my cart taxes should be :taxTotal
+     */
+    public function myCartTaxesShouldBe($taxTotal)
+    {
+        $this->summaryPage->open();
+
+        Assert::eq(
+            $this->summaryPage->getTaxTotal(),
+            $taxTotal,
+            'Tax total value should be %2$s, but it is %s.'
+        );
+    }
+
+    /**
+     * @Then shipping total value should be :shippingTotal
+     * @Then my cart shipping fee should be :shippingTotal
+     */
+    public function myCartShippingFeeShouldBe($shippingTotal)
+    {
+        $this->summaryPage->open();
+
+        Assert::eq(
+            $this->summaryPage->getShippingTotal(),
+            $shippingTotal,
+            'Shipping total value should be %2$s, but it is %s.'
+        );
+    }
+
+    /**
+     * @Then my cart promotions should be :promotionsTotal
+     * @Then my discount should be :promotionsTotal
+     */
+    public function myDiscountShouldBe($promotionsTotal)
+    {
+        $this->summaryPage->open();
+
+        Assert::eq(
+            $this->summaryPage->getPromotionTotal(),
+            $promotionsTotal,
+            'Promotion total value should be %2$s, but it is %s.'
+        );
+    }
+
+    /**
+     * @Given /^cart should be empty with no value$/
+     */
+    public function cartShouldBeEmptyWithNoValue()
+    {
+        $this->summaryPage->open();
+
+        expect($this->summaryPage)->toThrow(ElementNotFoundException::class)->during('getGrandTotal', []);
```

Comment:
I'm not convinced if exception driving development is a proper way. Maybe assert that some element does not exist would be a better solution?